### PR TITLE
fix(ingest): fix mysql ingestion issue with non-lowercase database

### LINKED ISSUE #6708: Failures with ingesting MySQL databases with names with upper case characters
**Describe the bug**
I get failures, see below, with the MySQL source after updating to Datahub v0.9.2 from v0.8.43. I get these failures where the database name has upper case characters. The name of the database where I get the failure below is really `amhDb`. I suspect that error was introduced when introducing `TwoTierSQLAlchemySource` in the `MySQLSource`. 

```
'failures': {'amhdb': ['Tables error: (pymysql.err.OperationalError) (1049, "Unknown database \'amhdb\'")\n[SQL: SHOW FULL TABLES FROM `amhdb`]\n(Background on this error at: https://sqlalche.me/e/14/e3q8)']},
```

**To Reproduce**
Ingest data with the mysql source where the database name has uppercase characters in the name.

**Expected behavior**
I do not expect any failure like in v0.8.43 when the database has a name with uppercase characters.

**Desktop (please complete the following information):**
 - Datahub version v0.9.2

**Additional context**
I suspect (I have not verified this) that the databases from `inspector.get_schema_names()` on line 84 are all in lower case and then when the new `url` is created on line 87 is defined with the name all in lower case and the error is introduced and I get failures when the inspector is used. 

https://github.com/datahub-project/datahub/blob/feb9d59b4d9c284bf4d8eaf83dd243d0bc8deeca/metadata-ingestion/src/datahub/ingestion/source/sql/two_tier_sql_source.py#L81-L91

### PR DESCRIPTION
**Pull request by mayurinehate**
Closes #6708 

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by treff7es — APPROVED** [review]
lgtm

**Review by treff7es — COMMENTED** [review]
@mayurinehate it seems like mysql intergration test is broken

File: metadata-ingestion/tests/integration/mysql/mysql_mces_no_db_golden.json [hunk 1]
```diff
@@ -315,7 +315,7 @@
 },
 {
     "entityType": "dataset",
-    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:mysql,datacharmer.salaries,PROD)",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:mysql,dataCharmer.salaries,PROD)",
```
Comment (mayurinehate) [inline metadata-ingestion/tests/integration/mysql/mysql_mces_no_db_golden.json:318]: Looking into ways to avoid change in dataset urn case for sql_common.

Comment (mayurinehate) [reply metadata-ingestion/tests/integration/mysql/mysql_mces_no_db_golden.json:318]: Current behavior looks good as schema level (the level immediately before table) has been case-sensitive for other sources as well

**Review by treff7es — APPROVED** [review]
lgtm

**Pull request merged**